server.js

```diff
@@ -18,9 +40,57 @@ const app = express()
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+// server ready
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).send({ error: 'service unavailable' })
+  }
+})
+
+// GET endpoints
+const listEndpoints = require('express-list-endpoints')
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+
+// GET Messages routes
+app.get('/thoughts', async (req, res) => {
+  try {
+    const messages = await Message.find()
+      .sort({ createdAt: 'desc' })
+      .limit(20)
+      .exec()
+    res.status(200).json(messages)
+  } catch (error) {
+    res.status(400).json({ message: "could not find messages", errors: err.errors })
+  }
+})
+
+// POST Message route
+app.post('/thoughts', async (req, res) => {
+  // send a request body in order to pass information into the API
+  try {
+    // success case
+    const NewMessage = new Message({ message: req.body.message, name: req.body.name })
+    const savedMessage = await NewMessage.save()
```

Comment:
This could also been written on just one line:
`const NewMessage = await new Message({ message: req.body.message, name: req.body.name }).save()`

And replace savedMessage on line 78 to newMessage